packages/sanity/src/core/studio/copyPaste/CopyPasteProvider.tsx

```diff
@@ -157,10 +136,13 @@ export const CopyPasteProvider: React.FC<{
   )
 
   const onPaste = useCallback(
-    async (targetPath: Path, value: FormDocumentValue | undefined, options?: PasteOptions) => {
-      const {schemaType: targetDocumentSchemaType} = documentMeta || {}
+    async (targetPath: Path, value: FormDocumentValue | undefined, options: PasteOptions) => {
+      // guard against `documentMeta` having not been set yet
+      if (!documentMeta) return
+
+      const {schemaType: targetDocumentSchemaType, onChange} = documentMeta
       const targetSchemaType = resolveSchemaTypeForPath(
```

Comment:
Should the toast and early return that guards against `targetDocumentSchemaType` happen _before_ this function is called?